src/formatDate.js

```diff
@@ -8,7 +8,53 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
```

Comment:
Using `pop()` on `fromFormat` and `toFormat` arrays to get the separators is problematic because it modifies the original arrays by removing the last element. This can lead to incorrect date mapping if the separators are not the last elements in the format arrays. Consider using a different approach to extract the separators without altering the arrays.